AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -31,13 +32,15 @@ class SyncMutationToCloudOperation: AsynchronousOperation {
     private var authTypesIterator: AWSAuthorizationTypeIterator?
 
     init(mutationEvent: MutationEvent,
+         getLatestSyncMetadata: @escaping () -> MutationSyncMetadata?,
```

Comment:
are there some benefits to passing in this closure over passing in the storageAdapter instance?

Reply:
The closure acting as an interface here. It could simplify testing. For current impelmentation, the `storageAdapter` is the hidden denpendency here. 

Btw, I realized the SQL query is not an async operation and I moved the get version logic to the very moment of creating the API request.